crates/polars-core/src/chunked_array/ops/repeat_by.rs

```diff
@@ -22,95 +22,127 @@ where
     fn repeat_by(&self, by: &IdxCa) -> PolarsResult<ListChunked> {
         check_lengths(self.len(), by.len())?;
 
-        if (self.len() != by.len()) & (by.len() == 1) {
-            return self.repeat_by(&IdxCa::new(
+        match (self.len(), by.len()) {
+            (left_len, right_len) if left_len == right_len => {
+                Ok(arity::binary(self, by, |arr, by| {
+                    let iter = arr.into_iter().zip(by).map(|(opt_v, opt_by)| {
+                        opt_by.map(|by| std::iter::repeat(opt_v.copied()).take(*by as usize))
+                    });
+
+                    // SAFETY: length of iter is trusted.
+                    unsafe {
+                        LargeListArray::from_iter_primitive_trusted_len(
```

Comment:
I believe we can share most of the code with these types, but before that, maybe we should unify this call first. 🤔

Reply:
Yes, might be a good follow up!